chunky/src/java/se/llbit/chunky/renderer/scene/Scene.java

```diff
@@ -1080,7 +1080,7 @@ public synchronized void loadChunks(TaskTracker taskTracker, World world, Collec
                       // Move plain water blocks to the water octree.
                       octNode = palette.airId;
 
-                      if(notOnEdge) {
+                      if(!onEdge) {
```

Comment:
Water finalization must also be performed if `y == yMin` or `y == yMax -1`. Only x/z edge (i.e. chunk crossing) is important here. This was the case before because `chunkData.isBlockOnEdge` (and `onEdge`) ignored the y coordinate.

Reply:
Should be good, it is done during the octree finalization, https://github.com/chunky-dev/chunky/blob/423ee0176b6b8da6d57166ca385d2fb78c1f5ec3/chunky/src/java/se/llbit/chunky/renderer/scene/OctreeFinalizer.java#L52

Reply:
Allright, maybe we could optimize this at some point in the future… Water finalization could happen here for y edge blocks too